### ISSUE TITLE
Changes

### ISSUE THREAD

**Issue by WayneManion** [davidmpye/Sonoff-Touch-MQTT#14]
Just FYI, at some point there appears to have been a hardware change on these things. I recently bought a Sonoff Touch that had an ESP8285 (an ESP8266 with integrated flash chip). If you select generic ESP8266 as the board type, the firmware won't work. 

The way the ESP module is placed on the board is different and the module is missing the antenna connector.

**Comment by davidmpye** [comment]
AFAIK, they've always been ESP8285s?


David

**Comment by WayneManion** [comment]
Yeah probably. After looking at more of your documentation, it looks like your pictures are not the US version of the Sonoff Touch like mine. 

On your ["How to flash" page](https://github.com/davidmpye/Sonoff-Touch-MQTT/wiki/How-to-flash):

"4) Using either the Arduino IDE or platform.io, upload the firmware. (If using arduino, 'Generic ESP8266 board' and Flash: (1M, 64K SPIFFS). If you don't get that part set correctly, it is likely OTA Upgrades will fail in future."

**Comment by davidmpye** [comment]
I'm sure I've been flashing mine with the Arduino IDE set to ESP8266 even
though it *is* an 8285.

But yeah, it seems inconsistent!

David

**Comment by Vanshava** [comment]
I encountered same issue that it bricked after flashing it according to the wiki document.
The board then cannot even boot up.... I cannot just simply re-flashing it again with ESP8285. 
May i have more information that how to re-flashing it?

------
Warning: espcomm_sync failed
error: espcomm_open failed
error: espcomm_upload_mem failed
error: espcomm_upload_mem failed

**Comment by davidmpye** [comment]
That error isn't terribly specific I am afraid, just that it won't
communicate with the uploader.

Check that your GPIO0 pulldown at boot was reliable....

David

**Comment by Vanshava** [comment]
GPIO0 grounded well as i managed flashing the custom firmware at beginning.
Completed flashing 100% and rebooted then it bricked. It no longer getting back into flashing mode again ever grounded GPIO0 again and It just bricked there... don't wanna talk with FDTI adapter ever.

Just googled "SONOFF bricked" seems i'm not alone. Still seeking some workaround...

**Comment by Nixsm** [comment]
Just managed to flash it correctly using the ESP8825 generic module in Arduino IDE

**Comment by davidmpye** [comment]
I wonder if it's something to do with me using platformio, rather than the
arduino IDE, but not entirely sure what.....
